docs/design/README.md

```diff
@@ -534,6 +534,149 @@ After installing packages, the result of `dnf list installed` will be output to
 This will allow maintainers to audit which kit each package was taken from.
 The rest of the variant image creation steps proceed the same way that they do now.
 
+### Build Algorithm Changes
+
+Currently, the build system relies on a flat directory of RPMs.
+It relies on Cargo to build packages so that dependent RPMs are in the directory when needed.
+When creating a variant image, all RPMs are available, but the yum install command can follow package dependencies to install the right things.
+
+When creating a kit, we won't have the yum install command to help us figure out all the right packages.
+We need to change the build system so that the dependency graph can be used to put the right packages into a kit.
+
+For starters, we will no longer rely on a flat structure in `build/rpms`.
+We will put the output of each `package` build into a subdirectory using the Cargo package's name.
+
+For example, when building `glibc/Cargo.toml` we will put the output in `build/rpms/glibc`.
+Note that more than one RPM file can result from a package build.
+The subdirectory maps the resultant RPM files to the Cargo package that produced them.
+
+At each `build-package` invocation, a distinct, flat RPM directory will be created containing the packages needed for the current `build-package` invocation.
+This will be done by traversing the Cargo dependency graph and making a list of packages and kits that are needed.
+The kits will be unpacked and the direct package dependencies will be copied from their `build/rpms` subdirectories.
+This will require us to pass a list of the packages needed for the build to the dockerfile, for example:
```

Comment:
We could leave kits in their unpacked form with loose RPMs + a repo definition, either instead of an archive or in addition to it.

It's sort of a trade-off between how complex we want the publish side to be vs. how complex we want the build side to be, since what's convenient for one is not for the other. But we could also just create both forms in the `build-kit` operation.